FusionAuth/fusionauth-site
# Track search matomo

This tracks the search results for matomo.

Note that if you want to test this locally, you need to disable the 

```
{% if jekyll.environment != 'development' %}
```

in `_includes/_head.liquid`.

Algolia didn't make it super easy to hook into where we get results (that I could find). I wanted a way I could:

* see how many results we got back (because matomo lets you send that info: https://developer.matomo.org/guides/tracking-javascript-guide#internal-search-tracking )
* see the query

I looked at a variety of docs:

* https://www.algolia.com/doc/guides/building-search-ui/going-further/integrate-google-analytics/js/
* https://www.algolia.com/doc/api-reference/widgets/insights/js/?client=ESM#widget-param-onevent
* https://www.algolia.com/doc/api-reference/widgets/analytics/js/

But couldn't find something that did everything I needed to.

So, instead of sending everything, I sent the query requested. Boo!

We'll also need to roll a new version of account when this is released.
